modules/tinymce/src/core/main/ts/selection/DetailsElement.ts

```diff
@@ -149,23 +151,20 @@ const preventDeleteIntoDetails = (editor: Editor, forward: boolean) => {
   if (editor.selection.isCollapsed()) {
     const caretPos = CaretPosition.fromRangeStart(selection.getRng());
     const parentBlock = dom.getParent(caretPos.container(), dom.isBlock);
-
-    // Prevent backspace/delete if the paragraph is empty and the start or end of it's parent container
-    // TODO: Clean this up!
-    if (parentBlock && dom.isEmpty(parentBlock)) {
-      if (Type.isNull(parentBlock.nextSibling)) {
-        const pos = CaretFinder.prevPosition(root, caretPos).filter((pos) => isInDetailsElement(dom, pos));
-        if (pos.isSome()) {
-          pos.each((pos) => {
-            if (!forward) {
-              moveCaretToDetailsPos(editor, pos);
-            }
-          });
-          return true;
-        }
-      } else if (Type.isNull(parentBlock.previousSibling)) {
-        const pos = CaretFinder.nextPosition(root, caretPos).filter((pos) => isInDetailsElement(dom, pos));
-        if (pos) {
+    const parentDetailsAtCaret = getParentDetailsElementAtPos(dom, caretPos);
+    const inEmptyParentBlock = parentBlock && dom.isEmpty(parentBlock);
+    const isFirstBlock = parentBlock && Type.isNull(parentBlock.previousSibling);
+    const isLastBlock = parentBlock && Type.isNull(parentBlock?.nextSibling);
```

Comment:
why is one row you use `parentBlock?` and in the row above only `parentBlock` without the `?`?

Reply:
Right I changed both to be null coalescing. I'm still getting used to use that feature of the language. :smile: